gradle-compat-6-5/src/main/java/wtf/emulator/GradleCompat_6_5.java

```diff
@@ -0,0 +1,28 @@
+package wtf.emulator;
+
+import org.gradle.api.Project;
+import org.gradle.api.internal.StartParameterInternal;
+import org.gradle.api.invocation.Gradle;
+import org.gradle.api.provider.Provider;
+
+import javax.annotation.Nullable;
+
+public class GradleCompat_6_5 extends GradleCompat_6_1 {
```

Comment:
Not sure how you feel about this approach where each compat extend the previous but it does reduce the amount of code.

Reply:
I think it's fine. Sometimes you can get class verification issues on loading if the superclass from an older version references types that aren't present in the newer one, but I think it's easier to break the extends chain whenever that happens.

Reply:
Good thing Gradle never removes anything 😂